app/models/work.rb

```diff
@@ -28,6 +28,12 @@ class Work < ApplicationRecord
   }
 
   state_machine initial: :first_draft do
+    after_transition deposited: :version_draft do |work, _transition|
+      Event.create!(work: work, user: work.depositor, event_type: 'new_version')
```

Comment:
Should `new_version` also be changed to `update_metadata` here?

Reply:
I don't think so, because new_version is really a name of a transition that happened.  Some update_metadata actions trigger that transition, but not all of them.

Reply:
Let's chat about this today.